core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -1707,4 +1707,38 @@ class KafkaApisTest {
       0, 0, partitionStates.asJava, Seq(broker).asJava).build()
     metadataCache.updateMetadata(correlationId = 0, updateMetadataRequest)
   }
+
+  @Test
+  def testAlterReplicaLogDirs(): Unit = {
+    val data = new AlterReplicaLogDirsRequestData()
+    val dir = new AlterReplicaLogDirsRequestData.AlterReplicaLogDir()
+      .setPath("/foo")
+    dir.topics().add(new AlterReplicaLogDirsRequestData.AlterReplicaLogDirTopic().setName("t0").setPartitions(asList(0, 1, 2)))
+    data.dirs().add(dir)
+    val alterReplicaLogDirsRequest = new AlterReplicaLogDirsRequest.Builder(
+      data
+    ).build()
+    val request = buildRequest(alterReplicaLogDirsRequest)
+
+    EasyMock.reset(replicaManager, clientRequestQuotaManager, requestChannel)
+
+    val capturedResponse = expectNoThrottling()
+    EasyMock.expect(replicaManager.alterReplicaLogDirs(EasyMock.eq(Map(
+      new TopicPartition("t0", 0) -> "/foo",
+      new TopicPartition("t0", 1) -> "/foo",
+      new TopicPartition("t0", 2) -> "/foo"))))
+      .andReturn(Map(
+        new TopicPartition("t0", 0) -> Errors.NONE,
+        new TopicPartition("t0", 1) -> Errors.LOG_DIR_NOT_FOUND,
+        new TopicPartition("t0", 2) -> Errors.INVALID_TOPIC_EXCEPTION))
+    EasyMock.replay(replicaManager, clientQuotaManager, clientRequestQuotaManager, requestChannel)
+
+    createKafkaApis().handleAlterReplicaLogDirsRequest(request)
+
+    val response = readResponse(ApiKeys.ALTER_REPLICA_LOG_DIRS, alterReplicaLogDirsRequest, capturedResponse)
+      .asInstanceOf[AlterReplicaLogDirsResponse]
+    assertEquals(Map(Errors.NONE -> 1,
```

Comment:
It is good to validate the `errorCounts` but could we also validate the overall mapping from TopicPartition to Errors?